### PR TITLE
Added a subscriber to the continuous hiking process (CHP) that will clear the waypoint list.

### PR DESCRIPTION
**Pull request by PotatoPeeler3000**

### CODE REVIEW

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/ReadyToPlanState.java [hunk 1]
```diff
@@ -263,4 +265,15 @@ public void addWayPointPoseToList(PoseListMessage poseListMessage)
       walkToGoalWayPointPoses.add(latestWayPoint);
       debugger.publishStartAndGoalForVisualization(continuousPlanner.getStartStancePose(), latestWayPoint);
    }
+
+   /**
+    * This allows the {@link ReadyToPlanState#walkToGoalWayPointPoses} to be cleared.
+    * This empties the list so the user can place a fresh goal that Continuous Hiking will use.
+    */
+   public void clearWayPointList(Empty emptyMessage)
```
Comment (PotatoPeeler3000) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/ReadyToPlanState.java:273]: This subscribes to a message that can be sent to the user. The JavaDoc should explain what this does. The reason for adding this now is that if the user mis-clicks there was nothing to clear that goal and reset. This caused problems when trying to use Continuous Hiking from Python.

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/ReadyToPlanState.java [hunk 2]
```diff
@@ -263,4 +265,15 @@ public void addWayPointPoseToList(PoseListMessage poseListMessage)
...
+   {
+      LogTools.info("Clearing waypoint list for WALK_TO_GOAL");
+      walkToGoalWayPointPoses.clear();
+      continuousHikingParameters.setEnableContinuousHiking(false);
```
Comment (PotatoPeeler3000) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/ReadyToPlanState.java:277]: Also if we are going to clear all the goals, we probably don't want to be walking, cause we have nothing to walk towards. So make sure to stop everything.

**Review by LuigiPenco93 — APPROVED** [review]

**Comment by rjgriffin42** [comment]
This is good. However, this should be squashed into a single commit, t here's only four lines of code that were changed. Fine if you do it by "squash and merge"

**Review by rjgriffin42 — APPROVED** [review]

**Comment by PotatoPeeler3000** [comment]
> This is good. However, this should be squashed into a single commit, t here's only four lines of code that were changed. Fine if you do it by "squash and merge"

Agreed, I haven't been considered the commit history too much but can start to consider it more.

**Pull request merged**